### DISCUSSION TITLE
[BUG] FHIR Sender: proxy and headers

### DISCUSSION THREAD

**Discussion by cldparisi** [nextgenhealthcare/connect#5355] [General]
I use “FHIR Sender” (with plugin FHIR on Mirth 3.10.1) but I have two problem:

-	It can’t use a proxy in “FHIR Sender” (like in “Web Service Sender”),  I can use it in “HTTP Sender” but not in FHIR and Web Sender.
Has anyone solved that problem?
is it possible set a proxy for “all Mirth” and use it  on all channels? 

-	It can’t add a headers, It’s possible to do it in “Web Service Sender” and “HTTP Sender” but it’s not possible in “FHIR sender”.
Is it possible to add headers by code before sending message?
is there another workaround?

Thanks

**Comment by pacmano1** [comment]
@cldparisi If you post here and the forums, please cross reference those.

Also, you are both asking questions (better for the discussion section) and asking for a feature, those two things should probably be two seperate things.   When you create a feature request it has a template you should follow.

I am converting this "bug" to a discussion for now.

**Reply by pacmano1** [reply]
You can add headers by exporting the channel and raw editing the XML and re-importing the channel, but it is not a great workaround.

The FHIR Sender is also not free, it is a commericial extenstion for which you should have support.  If you are not paying for it you are violating license terms.

Lastly, I am sure you know FHIR is just calling APIs, I'd just use an HTTP sender which does support all the options you want.

**Reply by cldparisi** [reply]
> @cldparisi If you post here and the forums, please cross reference those.

Ok.
https://forums.mirthproject.io/forum/mirth-connect/support/181879-fhir-sender-proxy-and-headers
 

> I am converting this "bug" to a discussion for now.

Ok, Thanks

**Reply by cldparisi** [reply]
> You can add headers by exporting the channel and raw editing the XML and re-importing the channel, but it is not a great workaround.

I've tried it before but it doesn't work (on Mirth 3.10.1)

 
> The FHIR Sender is also not free, it is a commericial extenstion for which you should have support. If you are not paying for it you are violating license terms.

FHIR Sender it is a commercial extension from 3.11 not previous versions, is it correct?

We are trying and testing it in 3.10.1 and then if it is ok for our needs we will evaluate the purchase for 4.0.0


> Lastly, I am sure you know FHIR is just calling APIs, I'd just use an HTTP sender which does support all the options you want.

Yes, infact we are trying both possibilities to decide which one to use in our projects

**Reply by pacmano1** [reply]
Actually Nextgen removed the "beta" version entirely and I think using even that older version is a violation of license terms.  I don't work for Nextgen, so it would be best to confirm with them.
